### DISCUSSION TITLE
Can sending of transactional email be offloaded on a separate thread?

### DISCUSSION THREAD

**Discussion by DDuran19** [pocketbase/pocketbase#5654] [Q&A] [answered]
Great app! 

I am using pocketbase as the backend of our e-commerce website.
so basically, everytime a customer creates an order, they will receive an email containing the order details.

It is working fine but it is really slow.. and I figured out that the order is waiting for the emails to be processed and sent before sending back the response back to the customer, a behavior that isn't desirable in our case..

also, if there was any sort of error on the sending of email part, the order is NOT cancelled but the entire response from the pocketbase fails.. leading for the user to make duplicate orders accidentally.. (cause they didn't know that the order pushed through)

so.. my question is, without extending using go..

Is it possible to offload the sending of email on a different process (like a cron job? I dont know, no idea) such that, the hook is only responsible for setting up the details needed for the email

Thank you and more powers!

Here's my code so far..

```js
// /pb_hooks/utils.js
module.exports = {
    sendEmail: (to, subject, html) => {
        const message = new MailerMessage({
            from: {
                address: $app.settings().meta.senderAddress,
                name: $app.settings().meta.senderName,
            },
            to: [{ address: to }],
            subject: subject,
            html: html,
        });

        return $app.newMailClient().send(message);
    },
};


//main.pb.js
/// <reference path="../pb_data/types.d.ts" />

// Hook for sending a confirmation email to the customer and admin when a new order is created
onRecordAfterCreateRequest((e) => {
    const utils = require(`${__hooks}/utils.js`);

    const id = e.record.id;
    const email = e.record.getString("email");
    const full_name = e.record.getString("full_name");
    const order_items = e.record.getString("order_items");
    const adminEmail = $app.settings().meta.senderAddress;

    // Prepare the order summary in HTML format
    const parsedOrderItems = JSON.parse(order_items);
    const cartHtml = parsedOrderItems.cart_items
        .map(
            (item) => `
        <div style="display: flex; justify-content: space-between; margin-bottom: 8px; font-size: 14px;">
          ...
        </div>
      `
        )
        .join("");

    const customerHtml = `
  <div style="font-family: Arial, sans-serif; color: hsl(220.9, 39.3%, 11%);">
    ...
  </div>
  `;

    const adminHtml = `
  <p>...</p>
  `;

    utils.sendEmail(email, "Order Confirmation", customerHtml);
    utils.sendEmail(adminEmail, "New Order Notification", adminHtml);
}, "online_orders");

// Hook for sending email updates to the customer when the order status changes
onRecordAfterUpdateRequest((e) => {
    const utils = require(`${__hooks}/utils.js`);
    const email = e.record.getString("email");
    const full_name = e.record.getString("full_name");
    const status = e.record.getString("status");

    const statusHtml = `
<div style="font-family: Arial, sans-serif; color: hsl(220.9, 39.3%, 11%);">
...
</div>
`;

    // Send status update email to the customer
    utils.sendEmail(email, "Order Status Update", statusHtml);
}, "online_orders");

```

**Comment by ganigeorgiev** [comment, marked as answer]
You can write the email data in a separate collection and then register a cron job that will read the created records from it and send the required email(s).

You can find a cron example in https://github.com/pocketbase/pocketbase/discussions/5235.

**Reply by DDuran19** [reply]
Thanks a lot ,  master! Its's brilliant. Can I set it to like, every minute? Or will it use too much resources even though there's nothing to send? Im new to this so bear with me :) if I extended it with go do I have to change the .exe? Cause im using pockethost.io right now, I know it isnt yours but Im not sure if they allow changing of the .exe file
